### PR TITLE
2383 - Admin users list improvements

### LINKED ISSUE #2383: Admin: Users list improvements
- [x] Add search by name (on top of roles)
- [x] Country column: show by default max 4 countries with show more/less button. e.g. Burundi, Central African Republic, Cameroon,...show more
- [x] On page change, scroll on top
- [x] verify count is really working. currently admin page differs. probably to @minotogna change in getMany
- [x] on cycle update, reset user store 
- [x] on cycle update, if user is in admin page, remain on admin page and refetch user list

### PR DESCRIPTION
**Pull request by rfontanarosa**
close #2383

https://user-images.githubusercontent.com/10047945/225938922-a3c557d2-8561-468f-8114-6fde21f8b32a.mov

### CODE REVIEW

File: src/client/components/UserListFilters/UserListFilters.tsx [hunk 1]
```diff
@@ -24,13 +24,36 @@ const UserListFilters: React.FC = () => {
 
   const [modalOpen, setModalOpen] = useState(false)
 
+  const [userName, setUserName] = useState('')
+
+  useEffect(() => {
+    const delayDebounceFn = setTimeout(() => {
```
Comment (minotogna) [inline src/client/components/UserListFilters/UserListFilters.tsx]: Does this work like a real debounce or the action is dispatched after 1 second at each change ?
Would it make sense to dispatch the fetch instead with debounce ?(even when other filters change )

File: src/client/components/PageLayout/Header/CycleSwitcher/CycleSwitcher.tsx [hunk 2]
```diff
@@ -28,9 +29,10 @@ const CycleSwitcher = () => {
     (cycleName: string) => {
       if (countryIso)
         navigate(ClientRoutes.Assessment.Cycle.Country.Landing.getLink({ countryIso, assessmentName, cycleName }))
+      else if (isAdmin) navigate(ClientRoutes.Assessment.Cycle.Admin.Root.getLink({ assessmentName, cycleName }))
```
Comment (minotogna) [inline src/client/components/PageLayout/Header/CycleSwitcher/CycleSwitcher.tsx:32]: this should happen only if current page is admin

File: src/client/store/ui/userManagement/actions/getUsers.ts [hunk 3]
```diff
@@ -13,16 +17,30 @@ type Params = {
...
   offset?: number
   countries?: Array<CountryIso>
+  fullname?: string
   roles?: Array<RoleName>
   administrators?: boolean
 }
 
-export const getUsers = createAsyncThunk<Array<User>, Params>('userManagement/get/users', async (params) => {
-  const { countryIso } = params
+const debouncedGetUsers = Functions.debounce(
```
Comment (minotogna) [inline src/client/store/ui/userManagement/actions/getUsers.ts]: forse farei un throttle qui.

Comment (minotogna) [reply src/client/store/ui/userManagement/actions/getUsers.ts]: a 500 ms con heading e trailing a true

File: src/client/store/ui/userManagement/actions/getUsersCount.ts [hunk 4]
```diff
@@ -1,25 +1,42 @@
...
   roles?: Array<RoleName>
 }
 
-export const getUsersCount = createAsyncThunk<{ totals: number }, Params>(
-  'userManagement/get/usersCount',
-  async (params) => {
-    const { data } = await axios.get(ApiEndPoint.Admin.usersCount(), {
-      params,
-    })
+const debouncedGetUsersCount = Functions.debounce(
```
Comment (minotogna) [inline src/client/store/ui/userManagement/actions/getUsersCount.ts]: throttle come su

File: src/server/repository/public/user/getMany.ts [hunk 5]
```diff
@@ -19,12 +19,13 @@ export const getMany = async (
...
     offset?: number
     countries?: Array<CountryIso>
+    fullname?: string
     roles?: Array<RoleName>
     administrators?: boolean
   },
   client: BaseProtocol = DB
 ): Promise<Array<User>> => {
-  const { countryIso, assessment, cycle, limit, offset, countries, roles, administrators } = props
+  const { countryIso, assessment, cycle, limit, offset, countries, fullname, roles, administrators } = props
```
Comment (sorja) [inline src/server/repository/public/user/getMany.ts]: ok as is;
but i would write fullName and full_name

**Review by sorja — DISMISSED** [review]

**Review by minotogna — APPROVED** [review]

**Review by sorja — APPROVED** [review]

**Pull request merged**